probe-rs/src/probe/mod.rs

```diff
@@ -25,6 +25,21 @@ impl fmt::Display for WireProtocol {
     }
 }
 
+impl std::str::FromStr for WireProtocol {
+    type Err = String;
```

Comment:
What's the intended use for this? Would it make snese to use a `DebugProbeError`, or `probe_rs::Error` here?

Reply:
I use it in cargo flash to automatically parse the protocol from structopt. and since this might be useful for pthers I added it to probe-rs. I don't think using an explicit error makes much sense here. At least I cannot see it right now.